### PR TITLE
Fix 1137

### PR DESCRIPTION
**Pull request by mebbaid**
This PR handles limitations imposed by `Irrlicht` on the number of vertices per rendered mesh. 
Related issue #1137

### CODE REVIEW

**Review by S-Dafarra — APPROVED** [review]

**Comment by S-Dafarra** [comment]
I think there might be an issue with ``assimp`` on Windows
```
LINK : fatal error LNK1181: cannot open input file 'assimp.lib' [D:\a\idyntree\idyntree\build\src\visualization\idyntree-visualization.vcxproj]
```

File: src/visualization/CMakeLists.txt [hunk 1]
```diff
@@ -102,6 +102,11 @@ if(IDYNTREE_USES_MESHCATCPP)
     target_link_libraries(${libraryname} PRIVATE MeshcatCpp::MeshcatCpp)
 endif()
 
+if(IDYNTREE_USES_ASSIMP)
+    target_compile_definitions(${libraryname} PRIVATE IDYNTREE_USES_ASSIMP)
+    target_link_libraries(${libraryname} PRIVATE assimp)
```
Comment (traversaro) [inline src/visualization/CMakeLists.txt:107]:
```suggestion
    target_link_libraries(${libraryname} PRIVATE assimp::assimp)
```

See https://github.com/robotology/idyntree/blob/07fffef70504d7e610d9466b4c3d12fd8ce610fc/src/solid-shapes/CMakeLists.txt#L28 as a reference. The logic there is more complex, but fortunately as we dropped Ubuntu 18.04 (see https://github.com/robotology/robotology-superbuild/pull/1527), we can just use the simple logic of linking to `assimp::assimp`

**Comment by traversaro** [comment]
> I think there might be an issue with `assimp` on Windows
> 
> ```
> LINK : fatal error LNK1181: cannot open input file 'assimp.lib' [D:\a\idyntree\idyntree\build\src\visualization\idyntree-visualization.vcxproj]
> ```

I guess this will be fixed by https://github.com/robotology/idyntree/pull/1139/files#r1446594328 .

**Comment by traversaro** [comment]
It is not a big problem in this case, but in the future if you could avoid making big whitespace/indent changes like the one in https://github.com/robotology/idyntree/pull/1139/commits/2b644e9eef218144bdef82ca9ebca9934021e949 mixed with actual changes it would be great, as the big whitespace/indentation changes make the actual changes difficult to read.

In the merge, can I squash together the commits? Thanks?

**Comment by mebbaid** [comment]
> but in the future if you could avoid making big whitespace/indent changes like the one in https://github.com/robotology/idyntree/commit/2b644e9eef218144bdef82ca9ebca9934021e949 mixed with actual changes it would be great, as the big whitespace/indentation changes make the actual changes difficult to read.

yeah, i relied too much on vscode for formating. But should pay attention to that.

> In the merge, can I squash together the commits? Thanks!

yeah, thanks a lot!

File: src/visualization/CMakeLists.txt (hunk 1 above)
Comment (mebbaid) [reply src/visualization/CMakeLists.txt]: done in https://github.com/mebbaid/idyntree/commit/0b6c88044c4cfc31f6f09ebaf18851fe83ec6377

**Comment by traversaro** [comment]
> > but in the future if you could avoid making big whitespace/indent changes like the one in [2b644e9](https://github.com/robotology/idyntree/commit/2b644e9eef218144bdef82ca9ebca9934021e949) mixed with actual changes it would be great, as the big whitespace/indentation changes make the actual changes difficult to read.
> 
> yeah, i relied too much on vscode for formating. But should pay attention to that.

It is partially fault of iDynTree itself, we should just add a .clang-format file and live happy.

**Pull request merged**